File: 0078-subsets/0078-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res=[]
        def sol(idx,subset):
            if idx==len(nums):
                res.append(subset.copy())
                return
            subset.append(nums[idx])
            sol(idx+1,subset)
            subset.pop()
            sol(idx+1,subset)
        sol(0,[])
        return res

File: 0078-subsets/0078-subsets.cs
public class Solution {
    public IList<IList<int>> Subsets(int[] nums) {
        var res=new List<IList<int>>();
        var curr=new List<int>();
        void solve(int idx){
            if (idx==nums.Length){
                res.Add(new List<int>(curr));
                return;
            }
            curr.Add(nums[idx]);
            solve(idx+1);
            curr.RemoveAt(curr.Count-1);

            solve(idx+1);
        }
        solve(0);
        return res;
    }
}